src/lib/statistics/statistics_objects/equal_distinct_count_histogram.cpp

```diff
@@ -0,0 +1,220 @@
+#include "equal_distinct_count_histogram.hpp"
+
+#include <cmath>
+
+#include <memory>
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "generic_histogram.hpp"
+#include "resolve_type.hpp"
+#include "storage/segment_iterate.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+template <typename T>
+std::unordered_map<T, HistogramCountType> add_segment_to_value_distribution(
+    const BaseSegment& segment, std::unordered_map<T, HistogramCountType> value_distribution,
+    const HistogramDomain<T>& domain) {
+  segment_iterate<T>(segment, [&](const auto& iterator_value) {
+    if (iterator_value.is_null()) return;
+
+    if constexpr (std::is_same_v<T, pmr_string>) {
+      // Do "contains()" check first to avoid the string copy incurred by string_to_domain() if we can
+      if (domain.contains(iterator_value.value())) {
+        ++value_distribution[iterator_value.value()];
+      } else {
+        ++value_distribution[domain.string_to_domain(iterator_value.value())];
+      }
+    } else {
+      ++value_distribution[iterator_value.value()];
+    }
+  });
+
+  return value_distribution;
+}
+
+template <typename T>
+std::vector<std::pair<T, HistogramCountType>> value_distribution_from_column(const Table& table,
+                                                                             const ColumnID column_id,
+                                                                             const HistogramDomain<T>& domain) {
+  std::unordered_map<T, HistogramCountType> value_distribution_map;
```

Comment:
I took the liberty to just add that as a comment
```c++
  // TODO(anybody) If you want to look into performance, this would probably benefit greatly from monotonic buffer 
  //               resources.
  std::unordered_map<T, HistogramCountType> value_distribution_map;
```